src/main/java/com/google/cloud/teleport/v2/bigquery-gcs/src/main/java/com/google/cloud/teleport/v2/bigquerygcs/BigQueryToParquet.java

```diff
@@ -0,0 +1,261 @@
+        } catch (IOException e) {
+          LOG.error("Error connecting to BigQueryStorage API: " + e.getMessage());
+          throw new RuntimeException(e);
+        }
+      }
+  }
+
+    /** Factory to create ReadSessions. */
+    static class ReadSessionFactory {
+
+        /**
+         * Creates ReadSession for schema extraction.
+         * @param tableString String that represents table to export from.
+         * @param tableReadOptions TableReadOptions that specify any fields in the table to filter on.
+         * @return session ReadSession object that contains the schema for the export.
+         */
+        ReadSession create(
+                BigQueryStorageClient client,
+                String tableString,
+                TableReadOptions tableReadOptions) {
+            TableReference tableReference = BigQueryHelpers.parseTableSpec(tableString);
+            String parent = "projects/" + tableReference.getProjectId();
+
+            TableReferenceProto.TableReference storageTableRef =
+                    TableReferenceProto.TableReference.newBuilder()
+                            .setProjectId(tableReference.getProjectId())
+                            .setDatasetId(tableReference.getDatasetId())
+                            .setTableId(tableReference.getTableId())
+                            .build();
+
+            CreateReadSessionRequest.Builder builder =
+                    CreateReadSessionRequest.newBuilder()
+                            .setParent(parent)
+                            .setReadOptions(tableReadOptions)
+                            .setTableReference(storageTableRef);
+            try {
+                return client.createReadSession(builder.build());
+            } catch (InvalidArgumentException iae) {
+                LOG.error("Error creating ReadSession: " + iae.getMessage());
+                throw iae;
+            }
+        }
+    }
+
+    /**
+     * The {@link BigQueryToParquetOptions} class provides the custom execution options passed by the executor at the
+     * command-line.
+     */
+    public interface BigQueryToParquetOptions extends PipelineOptions {
+        @Description("BigQuery table to export from in the form <project>:<dataset>.<table>")
+        @Required
+        String getTableRef();
+
+        void setTableRef(String tableRef);
+
+        @Description("GCS bucket to export BigQuery table data to (e.g. gs://mybucket/folder/).")
+        @Required
+        String getBucket();
+
+        void setBucket(String bucket);
+
+        @Description("Optional: Number of shards for output file.")
+        @Default.Integer(0)
+        Integer getNumShards();
+
+        void setNumShards(Integer numShards);
+
+        @Description("Optional: Comma separated list of fields to select from the table.")
+        String getFields();
+
+        void setFields(String fields);
+    }
+
+    /**
+     * The {@link getTableSchema} method gets Avro schema for table using BigQueryStorage API.
```

Comment:
That link seems incorrect. It should be ClassName#methodName IMO. can you confirm?

Reply:
Confirmed and updated.